app/controllers/api/functions.php

```diff
@@ -795,6 +840,23 @@
             $live = false;
         }
 
+        $spec = Config::getParam('runtime-sizes')[$size] ?? [];
+
+        if (empty($spec) || empty($spec['memory']) || empty($spec['cpus'])) {
+            throw new Exception(Exception::FUNCTION_INVALID_RUNTIME_SIZE);
+        }
+
+        // Enforce Cold Start if spec limits change.
+        if (($function->getAttribute('cpus') !== $spec['cpus'] ||
+            $function->getAttribute('memory') !== $spec['memory']) && !empty($function->getAttribute('deployment'))) {
+            $executor = new Executor(App::getEnv('_APP_EXECUTOR_HOST'));
+            try {
+                $executor->deleteRuntime($project->getId(), $function->getAttribute('deployment'));
```

Comment:
Lets make task in backlog, this should be deploymentId. We should address it in 1.7

Reply:
Done